packages/react/src/components/TextInput/TextInput.js

```diff
@@ -63,22 +95,65 @@ const TextInput = React.forwardRef(function TextInput(
   const helperTextClasses = classNames(`${prefix}--form__helper-text`, {
     [`${prefix}--form__helper-text--disabled`]: other.disabled,
   });
-  const label = labelText ? (
-    <label htmlFor={id} className={labelClasses}>
-      {labelText}
-    </label>
-  ) : null;
+  const label = (() => {
```

Comment:
Is there any way we can share some of this similar (at a glance) logic between TextArea and TextInput?

Reply:
it may be worth exploring once password input is finalized since it shares even more with TextInput than TextArea